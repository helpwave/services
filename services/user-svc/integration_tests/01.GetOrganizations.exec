#!/bin/bash
echo "Exists, has members:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "ead68d1d-db6b-4cf0-b316-ac9064c66413"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetOrganization 2>&1 || echo ""

echo "Exists, has no members:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "792145d6-39ae-4596-95b3-b2fd907256e3"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetOrganization

echo "Does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "77e49c1c-340b-4921-b5b4-fffbb8d3df6d"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/GetOrganization 2>&1 || echo ""
