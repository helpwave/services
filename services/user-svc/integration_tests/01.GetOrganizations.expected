Exists, has members:
{
  "id": "ead68d1d-db6b-4cf0-b316-ac9064c66413",
  "longName": "Uniklinik Aachen",
  "shortName": "Aachen",
  "contactEmail": "testine.test@helpwave.de",
  "members": [
    {
      "userId": "51dce0ea-2e72-412a-988e-8e9089a68572"
    },
    {
      "userId": "8b5ff90a-7548-454e-9684-6f2c30677db5"
    }
  ]
}
Exists, has no members:
{
  "id": "792145d6-39ae-4596-95b3-b2fd907256e3",
  "longName": "Uniklinik Köln",
  "shortName": "Köln",
  "contactEmail": "martin.test@helpwave.de"
}
Does not exist:
ERROR:
  Code: InvalidArgument
  Message: id not found

